src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2791,7 +2791,22 @@ def add_cors(cmd, resource_group_name, name, allowed_origins, slot=None):
     if not configs.cors:
         configs.cors = CorsSettings()
     configs.cors.allowed_origins = (configs.cors.allowed_origins or []) + allowed_origins
-    result = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
+    try:
+        result = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
+    except Exception as ex:  # pylint: disable=broad-except
+        poll_url = ex.response.headers['Location'] if 'Location' in ex.response.headers else None
+        if ex.response.status_code == 202 and poll_url:
+            r = send_raw_request(cmd.cli_ctx, method='get', url=poll_url)
+            poll_timeout = time.time() + 60 * 2  # 2 minute timeout
+
+            while r.status_code != 200 and time.time() < poll_timeout:
+                time.sleep(5)
+                r = send_raw_request(cmd.cli_ctx, method='get', url=poll_url)
+            return r.cors    
+        else:
+            raise CLIError(ex)
```

Comment:
Can you turn this into a function and just have both `add_cors` and `remove_cors` call that function?

Reply:
Also, this polling logic is only applicable for function apps on container app environments. Can you make sure this polling logic only gets called for that? Otherwise, we should be making the existing `update_configuration` API call.

Reply:
yes i had same thoughts as well .. thank you i will do it